koap/src/jsMain/kotlin/adapter/AdaptedMessage.kt

```diff
@@ -0,0 +1,165 @@
+package com.juul.koap.adapter
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@ExperimentalStdlibApi
+@Serializable
+data class AdaptedMessage(
```

Comment:
It may be simpler to mirror the naming of the actual classes that are being adapted.
Using `as` with the `import` allows you to distinguish where needed.

Reply:
I have updated the `adapter` package to use this approach.